eZ/Publish/Core/FieldType/Keyword/KeywordStorage/Gateway/LegacyStorage.php

```diff
@@ -378,4 +387,54 @@ protected function deleteOrphanedKeywords()
 
         $deleteQuery->prepare()->execute();
     }
+
+    /**
+     * @param \eZ\Publish\Core\Persistence\Database\DatabaseHandler $dbHandler
+     * @param mixed $fieldId
+     * @param int $versionNo
+     * @return bool
+     */
+    private function publishedVersionForFieldExists(
+        DatabaseHandler $dbHandler,
+        $fieldId,
+        $versionNo
+    ) {
+        $selectQuery = $dbHandler->createSelectQuery();
+        $selectQuery->select($dbHandler->quoteColumn('id', 'ezcontentobject_attribute'))
+            ->from(
+                $dbHandler->quoteTable('ezcontentobject_attribute')
+            )
+            ->innerJoin(
+                $dbHandler->quoteTable('ezcontentobject_version'),
+                $dbHandler->quoteColumn('contentobject_id', 'ezcontentobject_attribute'),
+                $dbHandler->quoteColumn('contentobject_id', 'ezcontentobject_version')
+            )
+            ->where(
+                $selectQuery->expr->lAnd(
+                    $selectQuery->expr->eq(
+                        $dbHandler->quoteColumn('id', 'ezcontentobject_attribute'),
+                        $selectQuery->bindValue($fieldId, null, \PDO::PARAM_INT)
+                    ),
+                    $selectQuery->expr->neq(
+                        $dbHandler->quoteColumn('version', 'ezcontentobject_attribute'),
+                        $selectQuery->bindValue($versionNo, null, \PDO::PARAM_INT)
```

Comment:
It is the same, but the actual Content Object is referred only by `$fieldId` which corresponds to `ezcontentobject_attribute`.`id` and `ezcontentobject_version` doesn't have this field (it's joined with `ezcontentobject_attribute` via  `contentobject_id`).

Also, looking at db indices, using here `ezcontentobject_version`.`version` instead of `ezcontentobject_attribute`.`version` shouldn't make a difference in performance.